### ISSUE TITLE
auto-load openconfig package in container-vqfx

### ISSUE THREAD

**Issue by mwiget** [nre-learning/antidote#208]
Network agent comes pre-loaded in the 18.4 image, but openconfig package isn’t. Need to package and auto-load ` junos-openconfig-0.0.0.10-1-signed.tgz` at runtime:

```
--- JUNOS 18.4R1.8 built 2018-12-17 03:30:15 UTC
{master:0}
mwiget@vqfx4> request system software add junos-openconfig-x86-32-0.0.0.10-1.tgz no-validate
malformed package: /var/tmp/incoming-package.tgz

{master:0}
mwiget@vqfx4> request system software add junos-openconfig-0.0.0.10-1-signed.tgz no-validate
Verified junos-openconfig-0.0.0.10-1.tgz signed by PackageProductionEc_2018 method ECDSA256+SHA256
Adding junos-openconfig...
Mounted junos-openconfig package on /dev/md13...
Verified manifest signed by PackageProductionEc_2018 method ECDSA256+SHA256
Executing /packages/mnt/junos-openconfig-0.0.0.10-1/mount.post..
This package is not compatible with Junos 18.4
Saving package file in /var/sw/pkg/junos-openconfig-0.0.0.10-1-signed.tgz ...
Saving state for rollback ...

{master:0}
mwiget@vqfx4> show version detail |match openconfig
JUNOS Openconfig [0.0.0.10-1
```

**Comment by Mierdin** [comment]
I can install this myself but even after installing, I can't seem to do any openconfig provisioning.

**Comment by mwiget** [comment]
Hmm, there is a config knob to unhide openconfig schema:

```
edit]
mwiget@openjnpr-container-vmx_vmx1_1# show system
schema {
    openconfig {
        unhide;
    }
}

edit]
mwiget@openjnpr-container-vmx_vmx1_1# set openconfig-bgp:bgp global afi-safis ?
Possible completions:
> afi-safi             AFI,SAFI configuration available for the
                       neighbour or group
[edit]
mwiget@openjnpr-container-vmx_vmx1_1# set openconfig-bgp:bgp global afi-safis
```

But doing that on vqfx 18.4 doesn’t help yet. Need to investigate further:

```
master:0}[edit]
mwiget@vqfx4# show system
schema {
    openconfig {
        unhide;
    }
}

{master:0}[edit]
mwiget@vqfx4# set ope?
No valid completions
{master:0}[edit]
mwiget@vqfx4# 
```

**Comment by mwiget** [comment]
Tried telemetry streaming via gRPC and while the jtimon client can connect, it can’t subscribe to any openconfig paths. The sensors aren’t present:

```
JUNOS Packet Forwarding Engine Support (qfx-10-f) [18.4R1.8]
JUNOS Routing Software Suite [18.4R1.8]
JUNOS jsd [i386-18.4R1.8-jet-1]
JUNOS SDN Software Suite [18.4R1.8]
JUNOS Enterprise Software Suite [18.4R1.8]
JUNOS Web Management [18.4R1.8]
JUNOS py-base-i386 [18.4R1.8]
JUNOS py-extensions-i386 [18.4R1.8]
JUNOS Openconfig [0.0.0.10-1]

{master:0}
mwiget@vqfx4> show lldp neighbors
Local Interface    Parent Interface    Chassis Id          Port info          System Name
xe-0/0/0           -                   1e:dd:4e:ce:d2:5f   Interface 8865 as eth1 3e70788c06b6
xe-0/0/1           -                   8a:15:dc:d3:ac:db   Interface 8867 as eth1 1b87f68675d2

{master:0}
mwiget@vqfx4> show agent sensors

{master:0}
mwiget@vqfx4> show system processes extensive | match "na-|jsd|mgd-api|xmlproxyd|rpd"
 1884 root       4    0   143M 53548K kqread   0:00  0.00% rpd
 1884 root       4    0   143M 53548K kqread   0:00  0.00% rpd
 1915 nobody    40    0  2440K  1636K select   0:00  0.00% na-mqttd
 1914 root      40    0 18348K  9296K select   0:00  0.00% jsd
 1914 root      40    0 18348K  9296K select   0:00  0.00% jsd
 1914 root       4    0 18348K  9296K select   0:00  0.00% jsd
 1919 root      40    0 16788K  7760K select   0:00  0.00% na-grpcd
 1919 root      40    0 16788K  7760K ucond    0:00  0.00% na-grpcd
 1919 root       8    0 16788K  7760K nanslp   0:00  0.00% na-grpcd
 1916 root       4    0 11564K  7516K select   0:00  0.00% mgd-api
 1917 root       4    0 10032K  5596K select   0:00  0.00% xmlproxyd

{master:0}
```

config:

```
{master:0}
mwiget@vqfx4> show configuration system
services {
    extension-service {
        request-response {
            grpc {
                clear-text {
                    port 50051;
                }
                skip-authentication;
            }
        }
    }
}
schema {
    openconfig {
        unhide;
    }
}
```

The sensor config isn’t in the ephemeral database neither:

```
master:0}
mwiget@vqfx4> show ephemeral-configuration instance junos-analytics
## Last changed: 2019-03-26 11:18:42 UTC
services {
    analytics;
}
```

jtimon client config:

```
mwiget@cbm1:container-vqfx$ cat lldp.json

{
  "port": 50051,
  "host": "172.23.0.7",
  "cid": "unique-alphanumeric-id-for-each-client-1",
  "grpc" : {
    "ws" : 524288
  },
  "paths": [{
    "path": "/lldp",
    "freq": 1000
  }]
}
```

Launching jtimon client:

```
. . .
compression = none                                                                                                                                    │········
Connecting to 172.23.0.7:50051                                                                                                                        │········
gRPC headers from host 172.23.0.7:50051                                                                                                               │········
  init-response: [response { subscription_id: 1 } ]                                                                                                   │········
  content-type: [application/grpc]                                                                                                                    │········
  grpc-accept-encoding: [identity,deflate,gzip]                                                                                                       │········
Receiving telemetry data from 172.23.0.7:50051                                                                                                        │········
                       
```

But no data arrives. Looking at the na-grpcd log, it shows the path /lldp not being valid (no sensor):

```
Mar 26 11:32:18  #426# Assigning Internal Id 0 to path: "/lldp/" sample_frequency: 1000
Mar 26 11:32:18  #427# <configuration-json>{"configuration":{"services":{"analytics":{"export-profile":[{"name":"export_0","transport":"grpc"}],"senso
r":[{"name":"sensor_0","export-name":"export_0","reporting-rate":"1","resource":"/lldp/","subscription-id":"0"}]}}}}</configuration-json>
Mar 26 11:32:18  #428# Subscribing for  client-1-NA_periodic_subscriber
Mar 26 11:32:18  #429# setsockopt: Setting receive buffer size to 8388608 for client-1-NA_periodic_subscriber
Mar 26 11:32:18  #430# getsockopt: Socket receive buffer size is 8388608 for client-1-NA_periodic_subscriber
Mar 26 11:32:18  #431# MGD command failed (cmd = 0). Error code: 4. Error message:
Invalid System Resource Name: /lldp/
;invalid value;
Mar 26 11:32:18  #432# System add/modify failed for path: "/lldp/" sample_frequency: 1000
Mar 26 11:32:18  #433# Unassigning Internal Id 0
Mar 26 11:32:18  #434# Failed to add, internal subscription 0 ref count = 0
Mar 26 11:32:18  #435# System Accepted Paths->additional_config { }
Mar 26 11:32:18  #436# Final-path-list:
Mar 26 11:32:18  #437# Final-internal-subscription-ids-list for susbcription 1 :
Mar 26 11:32:18  #438# Subscription client-1-NA_periodic_subscriber MQTT unsubscribe path: /0/1/0
Mar 26 11:32:18  #439# Subscription client-1-NA_periodic_subscriber MQTT unsubscribe path: /0/1/1/#
```

**Comment by mwiget** [comment]
@Mierdin despite having loaded openconfig module on vqfx, rebooted the instance and successfully connected via jtimon, no telemetry streaming data is received. There is something wrong with how openconfig got installed and launched.

**Comment by Mierdin** [comment]
Meaning with that version of Juno's? Would we try a different base image?

**Comment by mwiget** [comment]
That is the newest version. Might be a day one issue on vqfx image. I reached out to a few folks to troubleshoot further. Wonder if it is worth investigating to add vMX. This would require huge pages plus either a password protected docker hub repo for the container image with Junos and config.

**Comment by mwiget** [comment]
Did some more digging: The openconfig schema‘s didn‘t actually load. 18.4 reported an incompatibility, but still shows the module under ‚show version‘. I successfully loaded junos-openconfig-0.0.0.10-1-signed.tgz on 18.1R1 and can configure with openconfig schema‘s.

```
JUNOS py-extensions-i386 [18.1R1.9]
JUNOS Openconfig [0.0.0.10-1]

{master:0}[edit]
root@53d550bae827# set openconfig-?   
Possible completions:
> openconfig-bgp:bgp   Top-level configuration and state for the BGP router
> openconfig-interfaces:interfaces  Top level container for interfaces, including configuration
                       and state data.
. . .
```

**Comment by mwiget** [comment]
See pull request https://github.com/nre-learning/antidote/pull/209 for adding openconfig.

**Comment by Mierdin** [comment]
Closed via #209
